ci/clang-format.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+  cat <<EOF
+Usage: $0 [-h] [--fix]
+
+  Format c/c++ source that's tracked by git. Exits with a non-zero return code
+  when formatting is applied.
+
+  -h     Print this message
+  --fix  Format files in-place
+EOF
+}
+
+fix=
+while [ "$#" -gt 0 ] ; do
+  case $1 in
+
+    -h)
+      usage
+      exit 1
+      ;;
+
+    --fix)
+      fix=1
+      ;;
+
+    *)
+      echo "Unrecognized option: $1"
+      echo
+      usage
+      exit 1
+      ;;
+
+  esac
+
+  shift
+done
+
+cd "$(dirname "${BASH_SOURCE[0]}")/.."
+
+failure=
+for file in $(git ls-files | grep '\.\(cpp\|h\)$'); do
+  if grep -F -x "$file" ci/clang-format.ignore > /dev/null; then
+    continue
+  fi
+
+  formatted="${file}.formatted"
+  clang-format "$file" > "$formatted"
+  if ! diff "$file" "$formatted" > /dev/null; then
```

Comment:
Can I suggest `if ! cmp -s "$file" "$formatted"; then` instead? It should be faster than `diff` since you don't care what the differences are, only if there are any. Of note, the `-s` flag to `cmp` is standardized in POSIX, but the `--quiet` and `--silent` synonyms are GNU extensions.

Reply:
So much better, thank you!